docs/resources/samples/performance.md

```diff
@@ -0,0 +1,151 @@
+
+This script selects 1K cells per chunk but you can override to test out how it works for you. It updates 100k rows with 6 columns of data. Run this on a blank sheet to examine.
+
+```ts
+function main(workbook: ExcelScript.Workbook) {
+  const sheet = workbook.getActiveWorksheet();
+
+  let data: (string | number | boolean)[][] = [];
+  // Number of rows in the random data (x 6 columns).
+  const sampleRows = 100000;
+
+  console.log(`Generating data...`)
+  // Dynamically generate some random data for testing purpose. 
+  for (let i = 0; i < sampleRows; i++) {
+    data.push([i, ...[getRandomString(5), getRandomString(20), getRandomString(10), Math.random()], "Sample data"]);
+  }
+
+  console.log(`Calling update range function...`);
+  const updated = updateRangeInChunks(sheet.getRange("B2"), data);
+  if (!updated) {
+    console.log(`Update did not take place or complete. Check and run again.`)
+  }
+
+  return;
+}
+
+function updateRangeInChunks(
+  startCell: ExcelScript.Range,
+  values: (string | boolean | number)[][],
+  cellsInChunk: number = 10000
+): boolean {
+
+  const startTime = new Date().getTime();
+  console.log(`Cells per chunk setting: ${cellsInChunk}`);
+  if (!values) {
+    console.log(`Invalid input values to update.`);
+    return false;
+  }
+  if (values.length === 0 || values[0].length === 0) {
+    console.log(`Empty data -- nothing to update.`);
+    return true;
+  }
+  const totalCells = values.length * values[0].length;
+
+  console.log(`Total cells to update in the target range: ${totalCells}`);
+  if (totalCells <= cellsInChunk) {
+    console.log(`No need to chunk -- updating directly`);
+    updateTargetRange(startCell, values);
+    return true;
+  }
+
+  const rowsPerChunk = Math.floor(cellsInChunk / values[0].length);
+  console.log("Rows per chunk: " + rowsPerChunk);
+  let rowCount = 0;
+  let totalRowsUpdated = 0;
+  let chunkCount = 0;
+
+  for (let i = 0; i < values.length; i++) {
+    rowCount++;
+    if (rowCount === rowsPerChunk) {
+      chunkCount++;
+      console.log(`Calling update next chunk function. Chunk#: ${chunkCount}`);
+      updateNextChunk(startCell, values, rowsPerChunk, totalRowsUpdated);
+      rowCount = 0;
+      totalRowsUpdated += rowsPerChunk;
+      console.log(`${((totalRowsUpdated / values.length) * 100).toFixed(1)}% Done`);
+
+    }
+  }
+  console.log(`Updating remaining rows -- last chunk: ${rowCount}`)
+  if (rowCount > 0) {
+    updateNextChunk(startCell, values, rowCount, totalRowsUpdated);
+  }
+
+  let endTime = new Date().getTime();
+  console.log(`Completed ${totalCells} cells update. It took: ${((endTime - startTime) / 1000).toFixed(6)} seconds to complete. ${((((endTime  - startTime) / 1000)) / cellsInChunk).toFixed(8)} seconds per ${cellsInChunk} cells-chunk.`);
+
+  return true;
+}
+
+/**
+ * A Helper function that computes the target range and updates. 
+ */
+
+function updateNextChunk(
```

Comment:
@sumurthy, Is this function here to trick the transpiler into not batching the update calls?